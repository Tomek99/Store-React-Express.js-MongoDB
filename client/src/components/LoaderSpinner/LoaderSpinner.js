import ClipLoader from "react-spinners/ClipLoader";
import PropTypes from "prop-types";
function LoaderSpinner({ loading }) {
  return (
    <ClipLoader
      color="var(--main-color"
      loading={loading}
      size={150}
      aria-label="Loading Spinner"
      data-testid="loader"
    />
  );
}
LoaderSpinner.propTypes = {
  loading: PropTypes.bool,
};
export default LoaderSpinner;
