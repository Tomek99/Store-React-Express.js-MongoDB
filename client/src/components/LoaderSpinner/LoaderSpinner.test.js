import { render } from "@testing-library/react";
import LoaderSpinner from "./LoaderSpinner";

test("should render LoaderSpinner component", () => {
  const loading = true;
  const { container } = render(<LoaderSpinner loading={loading} />);

  expect(container).toMatchSnapshot();
});
