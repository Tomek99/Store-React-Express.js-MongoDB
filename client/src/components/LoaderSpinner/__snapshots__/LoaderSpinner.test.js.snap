// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoaderSpinner component 1`] = `
<div>
  <span
    aria-label="Loading Spinner"
    data-testid="loader"
    style="width: 150px; height: 150px; border-radius: 100%; border: 2px solid; border-bottom-color: transparent; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
  />
</div>
`;
